core/src/data/rasterSource.cpp

```diff
@@ -37,37 +37,37 @@ std::shared_ptr<TileData> RasterSource::parse(const TileTask& _task, const MapPr
 
 }
 
+std::shared_ptr<TileTask> RasterSource::createTask(TileID _tileId) {
```

Comment:
It could first check if the texture is already available - Didn't came to add this yet